configure.ac

```diff
@@ -202,6 +202,61 @@ AS_CASE([$with_gc],
     )
 AC_MSG_RESULT([$with_gc])
 
+dnl
+dnl User setting: native thread-local storage (off by default)
+dnl
+
+AC_ARG_WITH([native-tls],
+    [AS_HELP_STRING([--with-native-tls],
+        [use native thread-local storage implementation])],
+        [],
+        [with_native_tls=no])
+AC_MSG_CHECKING([whether to use native tls])
+AS_IF([test "x$with_native_tls" == "xyes"], [
+        AC_COMPILE_IFELSE(
+            [AC_LANG_SOURCE([[
+#if defined(__APPLE__) && defined(__x86_64__)
+#error FAIL
```

Comment:
If you just want to check for macOS, then you can do it much simpler, as we do elsewhere in this file:
```
case "$host_os" in
  *darwin*)
    ...
    ;;
esac
```
Also, do you really want to limit this to x86_64 ? So x86_32, and in the future ARM, are OK?

Reply:
I'm actually special-casing this for 64-bit Darwin, yes. That is intended.

Reply:
Still shouldn't use a compilation check for that, just match the host string against `x86_64-apple-darwin`

Reply:
That makes sense. I'll look at that.